Content.Server/Paper/PaperSystem.cs

```diff
@@ -37,15 +39,34 @@ private void OnInit(EntityUid uid, PaperComponent paperComp, ComponentInit args)
             paperComp.Mode = PaperAction.Read;
             UpdateUserInterface(uid, paperComp);
 
-            if (TryComp<AppearanceComponent>(uid, out var appearance))
+            UpdateAppearance(uid, paperComp);
+        }
+
+        private void OnMapInit(EntityUid uid, PaperComponent paperComp, MapInitEvent args)
+        {
+            if (Loc.TryGetString(paperComp.Content, out var locString))
+                paperComp.Content = locString;
+
+            if (paperComp.Content.Length > paperComp.ContentSize)
             {
-                if (paperComp.Content != "")
-                    appearance.SetData(PaperVisuals.Status, PaperStatus.Written);
+                paperComp.Content = paperComp.Content[..paperComp.ContentSize];
+            }
+        }
```

Comment:
any reason this is done on mapinit instead of component init? from my understanding now it seems like paper's appearancea would be incorrect (since init will update it to be not written as content is not set, then mapinit later sets cotnent without updating data)